doc/aaa/RBAC_Phase2.md

```diff
@@ -0,0 +1,421 @@
+- Will rely on notification to update any change in the user information, privilege or resource information
+
+### 1.1.4 Warm Boot Requirements
+N/A
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+
+The Translib code (also in sonic-mgmt-framework) will be modified to support RBAC via Roles, rather than Groups. It will receive username data from the REST/gNMI NBIs and perform the role lookup for a given user.
+
+For user management, a service must run on the host to sync the Redis User DB with the Linux user database and vice-versa.
+
+### 1.2.2 Container
+SONiC Management Framework, gNMI Telemetry containers
+
+### 1.2.3 SAI Overview
+N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+Enterprise networks that enforce authentication for their management interfaces.
+
+## 2.2 Functional Description
+This feature enables authentication and Role-Based Access Control (RBAC) on the REST and gNMI programmatic interfaces that are provided by the SONiC Management Framework and Telemetry containers. With respect to authentication, these programmatic interfaces will support password-based authentication with tokens, and certificate-based authentication.
+
+Since the Klish CLI in the management framework communicates with the REST server in the back-end, the solution will also be extended to support REST authentication.
+
+RBAC will be enforced centrally in the management framework, so that users accessing the system through varying interfaces will be limited to the same, consistent set of operations and objects depending on their role. Users' roles will be mapped using Linux Groups.
+
+Users and their role (group) assignments may be managed via Linux tools or the NBIs.
+
+# 3 Design
+## 3.1 Overview
+(TODO/DELL: Draw a picture)
+
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+* **UserTable**
+
+  This table contains the username to role mapping needed for enforcing the authorization checks.  It has the following columns :
+    * *user* : This is the username being authorized. This is a string.
+    * *tenant* : This contains the tenant with which the user is associated. This is a string
+    * *role* : This specifies the role associated with the username in the tenant. This is a comma separated list of strings.
+      The UserTable is keyed on <***user, tenant***>.
+
+  **Note**: The UserTable will _not_ store users' salted+hashed passwords due to security concerns surrounding access restrictions to the DB; instead, that information will be maintained in `/etc/shadow` as per Linux convention.
+
+* **PrivilegeTable**
+
+  This table has provides the information about the type of operations that a particular role is authorized to perform. The authorization can be performed at the granularity of a feature, feature group, or the entire system. The table has the following columns :
+    * *role* : The role associated with the user that is being authorized. This is a string.
```

Comment:
Do we still want to keep this scheme? Or should we modify it based on what you had up in Section 1.1.1.5?

Reply:
And if we do modify it based on the above, have we thought about whether to map one role --> multiple YANG paths, or to map each YANG path to multiple role(s)?

Reply:
This might actually make more sense, rather than what I originally had in 1.1.1.5. I'll revisit that, since it seems easier to implement